clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -3536,6 +3563,101 @@ private void produceToLocalKafkaHelper(
+    final InternalAvroSpecificSerializer<GlobalRtDivState> serializer =
+        AvroProtocolDefinition.GLOBAL_RT_DIV_STATE.getSerializer(); // unfortunately, serializer is not thread-safe
+    final byte[] keyBytes = getGlobalRtDivKeyBytes(brokerUrl);
+    final PubSubTopicPartition topicPartition = previousMessage.getTopicPartition();
+    TopicType realTimeTopicType = TopicType.of(REALTIME_TOPIC_TYPE, brokerUrl);
+
+    // Snapshot the VT DIV + RT DIV (single broker URL) in preparation to be produced
+    PartitionTracker divClone = kafkaDataIntegrityValidatorForLeaders.cloneProducerStates(partition, brokerUrl);
+    Map<CharSequence, ProducerPartitionState> rtDiv = divClone.getPartitionStates(realTimeTopicType);
+
+    // Create GlobalRtDivState (RT DIV + latestOffset) which will be serialized into a byte array
+    ByteBuffer emptyBuffer = ByteBuffer.allocate(0);
+    final long offset = previousMessage.getPosition().getNumericOffset();
+    GlobalRtDivState globalRtDiv = new GlobalRtDivState(brokerUrl, rtDiv, offset, emptyBuffer);
+    byte[] valueBytes = ByteUtils.extractByteArray(serializer.serialize(globalRtDiv));
+    try {
+      valueBytes = compressor.get().compress(valueBytes);
+    } catch (IOException e) {
+      LOGGER.error("Failed to compress GlobalRtDivState. Will proceed without {} compression", compressionStrategy, e);
+    }
+
+    // Create PubSubMessage for the LeaderProducerCallback to enqueue the RT + VT DIV to the drainer
+    KafkaKey divKey = new KafkaKey(MessageType.GLOBAL_RT_DIV, keyBytes);
+    KafkaMessageEnvelope divEnvelope = getVeniceWriter(partitionConsumptionState).get()
+        .getKafkaMessageEnvelope(
+            MessageType.PUT,
+            false,
+            partition,
+            true,
+            DEFAULT_LEADER_METADATA_WRAPPER,
+            APP_DEFAULT_LOGICAL_TS);
+    divEnvelope.payloadUnion = new DivSnapshot(divClone, offset); // contains both VT + RT DIV
+    DefaultPubSubMessage divMessage = new ImmutablePubSubMessage(
+        divKey,
+        divEnvelope,
+        topicPartition,
+        previousMessage.getPosition(),
+        previousMessage.getPubSubMessageTime(),
+        divKey.getHeapSize() + valueBytes.length);
+    LeaderProducerCallback divCallback = createProducerCallback(
+        divMessage,
+        partitionConsumptionState,
+        leaderProducedRecordContext,
+        partition,
+        brokerUrl,
+        beforeProcessingRecordTimestampNs);
+
+    // Get the old value manifest which contains the list of old chunks, so they can be deleted
+    ChunkedValueManifestContainer valueManifestContainer = new ChunkedValueManifestContainer();
+    final int schemaId = AvroProtocolDefinition.GLOBAL_RT_DIV_STATE.getCurrentProtocolVersion();
+    readStoredValueRecord(partitionConsumptionState, keyBytes, schemaId, topicPartition, valueManifestContainer);
+
+    // Produce to local kafka for the Global RT DIV + latestOffset (GlobalRtDivState)
```

Comment:
Can we update the comments a bit. e.g.
// Produce to local kafka for the Global RT DIV + latestOffset (GlobalRtDivState)
// VeniceWriter.put() will schedule DELETEs for the old chunks in the old manifest after the new PUT